src/element/element_stream_writer.rs

```diff
@@ -0,0 +1,697 @@
+        // TODO what does this mean for an element writer?
+        todo!()
+    }
+
+    fn supports_text_symbol_tokens(&self) -> bool {
+        true
+    }
+
+    fn set_annotations<I, A>(&mut self, annotations: I)
+    where
+        A: AsRawSymbolTokenRef,
+        I: IntoIterator<Item = A>,
+    {
+        let annotations: IonResult<_> =
+            annotations.into_iter().map(|s| self.to_symbol(s)).collect();
+        // TODO what to do with the expect?
+        self.set_current_annotations(Annotations::new(annotations.expect("symbols")));
+    }
+
+    fn write_null(&mut self, ion_type: IonType) -> IonResult<()> {
+        self.write_scalar(Value::Null(ion_type))
+    }
+
+    fn write_bool(&mut self, value: bool) -> IonResult<()> {
+        self.write_scalar(Value::Bool(value))
+    }
+
+    fn write_i64(&mut self, value: i64) -> IonResult<()> {
+        self.write_scalar(Value::Int(Int::I64(value)))
+    }
+
+    fn write_int(&mut self, value: &Int) -> IonResult<()> {
+        self.write_scalar(Value::Int(value.clone()))
+    }
+
+    fn write_f32(&mut self, value: f32) -> IonResult<()> {
+        self.write_scalar(Value::Float(value as f64))
+    }
+
+    fn write_f64(&mut self, value: f64) -> IonResult<()> {
+        self.write_scalar(Value::Float(value))
+    }
+
+    fn write_decimal(&mut self, value: &Decimal) -> IonResult<()> {
+        self.write_scalar(Value::Decimal(value.clone()))
+    }
+
+    fn write_timestamp(&mut self, value: &Timestamp) -> IonResult<()> {
+        self.write_scalar(Value::Timestamp(value.clone()))
+    }
+
+    fn write_symbol<A: AsRawSymbolTokenRef>(&mut self, value: A) -> IonResult<()> {
+        let symbol = self.to_symbol(value)?;
+        self.write_scalar(Value::Symbol(symbol))
+    }
+
+    fn write_string<A: AsRef<str>>(&mut self, value: A) -> IonResult<()> {
+        self.write_scalar(Value::String(Str::from(value.as_ref())))
+    }
+
+    fn write_clob<A: AsRef<[u8]>>(&mut self, value: A) -> IonResult<()> {
+        self.write_scalar(Value::Clob(Bytes::from(value.as_ref())))
+    }
+
+    fn write_blob<A: AsRef<[u8]>>(&mut self, value: A) -> IonResult<()> {
+        self.write_scalar(Value::Blob(Bytes::from(value.as_ref())))
+    }
+
+    fn step_in(&mut self, ion_type: IonType) -> IonResult<()> {
+        let container = match ion_type {
+            IonType::Struct => Container::Struct(vec![]),
+            IonType::List => Container::List(vec![]),
+            IonType::SExp => Container::SExpression(vec![]),
+            _ => return illegal_operation(format!("Cannot step into a(n) {ion_type:?}")),
+        };
+        self.push_container(container)
+    }
+
+    fn set_field_name<A: AsRawSymbolTokenRef>(&mut self, name: A) {
+        // TODO what to do with the panic?
+        match self.to_symbol(name) {
+            Ok(s) => self.field_name = Some(s),
+            Err(_e) => panic!("set field name invalid symbol"),
+        }
+    }
```

Comment:
This is one place that symbol ids could be passed to `to_symbol`, but this interface method is not fallible.

Reply:
Same thought as above--attempt to resolve; otherwise panic and link to #564.

Reply:
Done